### PR TITLE
Implement Try runtime

### PR DESCRIPTION
**Pull request by pifragile**
Needed for https://github.com/encointer/encointer-parachain/issues/176 in order to test https://github.com/encointer/encointer-parachain/issues/175

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -10,27 +10,27 @@ members = [
...
-# pallet-encointer-ceremonies-rpc-runtime-api = { path = "../encointer-pallets/ceremonies/rpc/runtime-api" }
-# pallet-encointer-communities = { path = "../encointer-pallets/communities" }
-# pallet-encointer-communities-rpc = { path = "../encointer-pallets/communities/rpc" }
-# pallet-encointer-communities-rpc-runtime-api = { path = "../encointer-pallets/communities/rpc/runtime-api" }
-# pallet-encointer-balances = { path = "../encointer-pallets/balances" }
-# pallet-encointer-scheduler = { path = "../encointer-pallets/scheduler" }
-# pallet-encointer-bazaar = { path = "../encointer-pallets/bazaar" }
-# pallet-encointer-bazaar-rpc = { path = "../encointer-pallets/bazaar/rpc" }
-# pallet-encointer-bazaar-rpc-runtime-api = { path = "../encointer-pallets/bazaar/rpc/runtime-api" }
+[patch."https://github.com/encointer/pallets"]
```
Comment (clangenb) [inline Cargo.toml]: you need to remove the patches.

File: node/Cargo.toml [hunk 2]
```diff
@@ -47,6 +47,7 @@ sp-runtime = { version = "7.0.0", git = "https://github.com/paritytech/substrate
 sp-timestamp = { version = "4.0.0-dev", git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.37" }
 sp-inherents = { version = "4.0.0-dev", git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.37" }
 sp-keyring = { version = "7.0.0", git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.37" }
+sp-io = { version = "7.0.0", git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.37" }
```
Comment (clangenb) [inline node/Cargo.toml:48]: I don't see why this should be needed now?

File: client/src/community_spec.rs [hunk 3]
```diff
@@ -125,7 +131,7 @@ pub fn new_community_call<T: CommunitySpec>(spec: &T, metadata: &Metadata) -> Ne
 	info!("Metadata: {:?}", meta);
 
 	info!("bootstrappers: {:?}", bootstrappers);
-	info!("name: {}", meta.name);
+	info!("name: {:?}", meta.name);
```
Comment (clangenb) [inline client/src/community_spec.rs:155]: why?

File: node/Cargo.toml [hunk 4]
```diff
@@ -79,7 +80,8 @@ encointer-balances-tx-payment-rpc-runtime-api = { git = "https://github.com/enco
 encointer-balances-tx-payment-rpc = { git = "https://github.com/encointer/pallets", branch = "polkadot-v0.9.37" }
 
 # CLI-specific dependencies
-try-runtime-cli = { version = "0.10.0-dev", optional = true, git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.37" }
+frame-try-runtime = { optional = true, git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.37" }
+try-runtime-cli = { optional = true, git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.37" }
```
Comment (clangenb) [inline node/Cargo.toml:84]: if the version has a `-dev` suffix, we need to explicitly mention it.

File: runtime/Cargo.toml [hunk 5]
```diff
@@ -54,6 +54,10 @@ sp-std = { version = "5.0.0", default-features = false, git = "https://github.co
 sp-transaction-pool = { version = "4.0.0-dev", default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.37" }
 sp-version = { version = "5.0.0", default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.37" }
 
+frame-try-runtime = { git = 'https://github.com/paritytech/substrate.git', default-features = false, branch = "polkadot-v0.9.37", optional = true }
+try-runtime-cli = { git = 'https://github.com/paritytech/substrate.git', default-features = false, branch = "polkadot-v0.9.37", optional = true }
```
Comment (clangenb) [inline runtime/Cargo.toml]: please adhere to convention:

```
{ version = , default-features = false, optional = true, git =..., branch = ... }
```

File: runtime/Cargo.toml [hunk 6]
```diff
@@ -127,3 +133,27 @@ runtime-benchmarks = [
...
+	"pallet-scheduler/try-runtime",
+	"pallet-utility/try-runtime",
+    "pallet-asset-tx-payment/try-runtime",
+
+    "pallet-encointer-balances/try-runtime",
+    "pallet-encointer-bazaar/try-runtime",
+    "pallet-encointer-ceremonies/try-runtime",
+    "pallet-encointer-communities/try-runtime",
+    "pallet-encointer-scheduler/try-runtime",
+]
```
Comment (clangenb) [inline runtime/Cargo.toml]: fmt

**Review by clangenb — COMMENTED** [review]
Some toml comments, but it looks good otherwise.

File: client/src/community_spec.rs (hunk 3 above)
Comment (pifragile) [reply client/src/community_spec.rs:155]: because it is a `BoundedVec` now

File: Cargo.toml (hunk 1 above)
Comment (pifragile) [reply Cargo.toml]: i left the patches for easier testing. did you already test it?

File: node/Cargo.toml (hunk 4 above)
Comment (pifragile) [reply node/Cargo.toml:84]: i cannot follow.

**Review by clangenb — COMMENTED** [review]
Can you tell me how to test it?

**Comment by pifragile** [comment]
> Can you tell me how to test it?

Check https://github.com/encointer/encointer-parachain/pull/177

**Review by clangenb — APPROVED** [review]
Wuhuu, ready to merge!

**Pull request merged**